codegen/lib/graphql_java_gen/templates/APISchema.java.erb

```diff
@@ -211,14 +212,14 @@ public class <%= schema_name %> {
                       children.add(this);
                     <% end %>
                     <% type.fields.each do |field| %>
-                      <% next unless field.type.unwrap.object? %>
+                      <% next unless field.type.unwrap.object? || field.type.unwrap.interface? %>
                       if (get<%= field.classify_name %>() != null) {
                         <% if field.type.unwrap_non_null.kind == 'LIST' %>
                             for (<%= java_output_type(field.type.unwrap) %> elem: get<%= field.classify_name %>()) {
-                              children.addAll(elem.getNodes());
+                              children.addAll(elem.self().getNodes());
```

Comment:
Couldn't you just cast the value to AbstractResponse to avoid adding a `self` method?